addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXGenericThingHandler.java

```diff
@@ -0,0 +1,1124 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.knx.handler;
+
+import static org.openhab.binding.knx.KNXBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.openhab.binding.knx.GroupAddressListener;
+import org.openhab.binding.knx.IndividualAddressListener;
+import org.openhab.binding.knx.internal.channel.KNXChannelSelectorProxy;
+import org.openhab.binding.knx.internal.channel.KNXChannelSelectorProxy.KNXChannelSelector;
+import org.openhab.binding.knx.internal.dpt.KNXCoreTypeMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tuwien.auto.calimero.DataUnitBuilder;
+import tuwien.auto.calimero.DeviceDescriptor;
+import tuwien.auto.calimero.DeviceDescriptor.DD0;
+import tuwien.auto.calimero.GroupAddress;
+import tuwien.auto.calimero.IndividualAddress;
+import tuwien.auto.calimero.datapoint.CommandDP;
+import tuwien.auto.calimero.datapoint.Datapoint;
+import tuwien.auto.calimero.exception.KNXFormatException;
+import tuwien.auto.calimero.mgmt.PropertyAccess.PID;
+
+/**
+ * The {@link KNXGenericThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public class KNXGenericThingHandler extends BaseThingHandler
+        implements IndividualAddressListener, GroupAddressListener {
+
+    protected Logger logger = LoggerFactory.getLogger(KNXGenericThingHandler.class);
+
+    protected KNXChannelSelectorProxy knxChannelSelectorProxy = new KNXChannelSelectorProxy();
+
+    // the physical address of the KNX actor represented by this Thing
+    protected IndividualAddress address;
+
+    // group addresses the handler is monitoring
+    protected Set<GroupAddress> groupAddresses = new HashSet<GroupAddress>();
```

Comment:
This would limit the functionality of classes inheriting from KNXGenericThingHandler